parity/db/rocksdb/helpers.rs

```diff
@@ -29,32 +29,32 @@ pub fn compaction_profile(profile: &DatabaseCompactionProfile, db_path: &Path) -
 
 /// Spreads the `total` (in MiB) memory budget across the db columns.
 /// If it's `None`, the default memory budget will be used for each column.
-pub fn memory_per_column(total: Option<usize>) -> HashMap<Option<u32>, usize> {
+pub fn memory_per_column(total: Option<usize>) -> HashMap<u32, usize> {
 	let mut memory_per_column = HashMap::new();
 	if let Some(budget) = total {
 		// spend 90% of the memory budget on the state column, but at least 256 MiB
 		memory_per_column.insert(ethcore_db::COL_STATE, std::cmp::max(budget * 9 / 10, 256));
-		let num_columns = ethcore_db::NUM_COLUMNS.expect("NUM_COLUMNS is Some; qed");
 		// spread the remaining 10% evenly across columns
-		let rest_budget = budget / 10 / (num_columns as usize - 1);
-		for i in 1..num_columns {
+		let rest_budget = budget / 10 / (ethcore_db::NUM_COLUMNS as usize - 1);
+		// todo[dvdplm] relies on `COL_STATE` being the first column?
+		for i in 1..ethcore_db::NUM_COLUMNS {
 			// but at least 16 MiB for each column
-			memory_per_column.insert(Some(i), std::cmp::max(rest_budget, 16));
+			memory_per_column.insert(i, std::cmp::max(rest_budget, 16));
 		}
 	}
 	memory_per_column
 }
 
 /// Spreads the `total` (in MiB) memory budget across the light db columns.
-pub fn memory_per_column_light(total: usize) -> HashMap<Option<u32>, usize> {
+pub fn memory_per_column_light(total: usize) -> HashMap<u32, usize> {
 	let mut memory_per_column = HashMap::new();
-	let num_columns = ethcore_db::NUM_COLUMNS.expect("NUM_COLUMNS is Some; qed");
 	// spread the memory budget evenly across columns
 	// light client doesn't use the state column
-	let per_column = total / (num_columns as usize - 1);
-	for i in 1..num_columns {
+	let per_column = total / (ethcore_db::NUM_COLUMNS as usize - 1);
+	// todo[dvdplm] doesn't `1..` mean that col0 will get the default cache size assigned?
```

Comment:
yes, but we're not using it in the light client (and it doesn't allocate memory if not used)

Reply:
I think it would be better to be explicit here. It's not fair to assume the reader knows the inner details of the light impl; and I guess the same is true for rocksdb's allocation policy.

Reply:
Agree, especially, it's not clear why we pass `cache_config.blockchain()` as `total`, instead of `db_cache_size`